specs/agents/tracing-instrumentation-aws-lambda.md

```diff
@@ -0,0 +1,143 @@
+
+- API Gateway V1 
+- API Gateway V2 
+- SQS
+- SNS
+- S3
+
+If none of the above apply, the fallback should be a generic instrumentation (as described above) that can deal with any type of trigger (thus capturing only the minimal available information).
+
+### API Gateway (V1 & V2)
+There are two different API Gateway versions (V1 & V2) that differ slightly in the information (`event` object) that is passed to the Lambda handler function. 
+
+With both versions, the `event` object contains information about the http request.
+Usually API Gateway-based Lambda functions return an object that contains the HTTP response information.
+The agent should use the information in the request and response objects to fill the HTTP context (`context.request` and `context.response`) fields in the same way it is done for HTTP transactions.
+
+In particular, agents must use the `event.headers` to retrieve the `traceparent` and the `tracestate` and use them to start the transaction for the lambda function execution. 
+
+In addition the following fields should be set for API Gateway-based Lambda functions:
+
+Field | Value | Description | Source
+---   | ---   | ---         | ---
+`faas.trigger.type` | `http` | Constant value for API gateway. | -
+`transaction.name` | e.g. `GET MyFunction` | Http method followed by a whitespace and the function name. | - 
+`faas.trigger.reuqest_id` | e.g. `afa4-a6...` | ID of the API gateway request. | `event.requestContext.requestId` 
+`service.origin.name` | e.g. `POST /{proxy+}/Prod` | Readable API gateway endpoint. |Format: `${event.requestContext.httpMethod} ${event.requestContext.resourcePath}/${event.requestContext.stage}` 
+`service.origin.id` | e.g. `gy415nu...` | `event.requestContext.apiId` | 
+`service.origin.version` | e.g. `1.0` | `1.0` for API Gateway V1, `2.0` for API Gateway V2. | -
+`cloud.origin.service.name` | `api gateway` | Fix value for API gateway. | -
+`cloud.origin.account.id` | e.g. `12345678912` | Account ID of the API gateway. | `event.requestContext.accountId`
+
+### SQS / SNS
+Lambda functions that are triggered by SQS (or SNS) accept an `event` input that may contain one ore more SQS / SNS messages in the `event.records` array. All message-related context information (including the `traceparent`) is encoded in the individual message attributes (if at all). We cannot (automatically) wrap the processing of the individual messages that are sent as a batch of messages with a single `event`. 
+
+Thus, in case that an SQS / SNS `event` contains **exactly one** SQS / SNS message, the agents must apply the following, messaging-specific retrieval of information. Otherwise, the agents should apply the [Generic Lambda Instrumentation](generic-lambda-instrumentation) as desribed above.
```

Comment:
```suggestion
Thus, in case that an SQS / SNS `event` contains **exactly one** SQS / SNS message, the agents must apply the following, messaging-specific retrieval of information. Otherwise, the agents should apply the [Generic Lambda Instrumentation](generic-lambda-instrumentation) as described above.
```